### DISCUSSION TITLE
BasePath attribute

### DISCUSSION THREAD

**Discussion by aleksander-ivanov** [reactiveui/refit#1627] [General]
Does Refit has [BasePath] attribute [RestEase](https://github.com/canton7/RestEase#base-path) analogy?

**Comment by glennawatson** [comment]
Refit has no [BasePath] attribute like RestEase. The base path comes from the HttpClient.BaseAddress, and Refit does honour the path portion of it.

So set the common prefix on the client:

```csharp
// DI
builder.Services.AddRefitClient<IMyApi>()
    .ConfigureHttpClient(c => c.BaseAddress = new Uri("https://api.example.com/v1/"));

// or manual
var client = new HttpClient { BaseAddress = new Uri("https://api.example.com/v1/") };
var api = RestService.For<IMyApi>(client);
```

Then your method paths are relative to that:

```csharp
[Get("/users/{id}")] // resolves under /v1
Task<User> GetUser(int id);
```

Notes:
- Keep a trailing slash on the BaseAddress path and use relative method paths for predictable combination.
- If you want a per-method shared prefix without repeating it, factor it into the BaseAddress (or split into separate interfaces/clients). There is no attribute-based base path token like RestEase's {BasePath}.
